x-pack/plugins/maps/public/actions/store_actions.js

```diff
@@ -180,26 +184,24 @@ export function setSelectedLayer(layerId) {
   };
 }
 
-export function updateLayerOrder(newLayerOrder) {
-  return {
-    type: UPDATE_LAYER_ORDER,
-    newLayerOrder
+export function setTransientLayer(layerId) {
+  return  {
+    type: SET_TRANSIENT_LAYER,
+    transientLayerId: layerId,
   };
 }
 
-export function promoteTemporaryLayers() {
-  return {
-    type: PROMOTE_TEMPORARY_LAYERS
+export function saveWorkspace() {
```

Comment:
This action-creator isn't really saving anything, so I'd rename this to what is actually does. e.g. `clearTransientLayerStateAndCloseFlyout`.